src/librustc_data_structures/vob_map_set.rs

```diff
@@ -0,0 +1,247 @@
+use vob::{self, Vob};
+use std::iter::Iterator;
+
+/// Conceptually this is a `Map<usize, Set<usize>>`, using a single bit vector internally.
+///
+/// * The size of the sets for each key must be the same and known ahead of time.
+/// * Operations will panic if you try to access out of bounds keys or set items.
+/// * When the map is resized, new keys will be instantiated with empty sets. As such, this data
+///   structure cannot reliably express the absence of a key.
+pub struct VobMapSet {
+    vob: Vob,
+    set_size: usize,
```

Comment:
Isn't this always the same size as `vob.len()`?

Reply:
No. It's the size of the set for each key. `vob.len()` will be this number multiplied by the number of keys.

Reply:
OK.